packages/commonwealth/client/scripts/state/api/comments/deleteReaction.ts

```diff
@@ -0,0 +1,85 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import axios from 'axios';
+import ReactionCount from 'models/ReactionCount';
+import app from 'state';
+import { ApiEndpoints } from 'state/api/config';
+import useFetchCommentReactionsQuery from './fetchReaction';
+
+interface DeleteReactionProps {
+  reactionCount: ReactionCount<any>
+  canvasHash: string
+  reactionId: number;
+}
+
+const deleteReaction = async ({
+  reactionCount,
+  canvasHash,
+  reactionId
+}: DeleteReactionProps) => {
+  const {
+    session = null,
+    action = null,
+    hash = null,
+  } = await app.sessions.signDeleteCommentReaction({
+    comment_id: canvasHash,
+  })
+
+  return await axios.delete(`${app.serverUrl()}/reactions/${reactionId}`, {
+    data: {
+      jwt: app.user.jwt,
+      canvas_action: action,
+      canvas_session: session,
+      canvas_hash: hash,
+    },
+  }).then((r) => ({
+    ...r,
+    data: {
+      ...r.data,
+      result: {
+        ...(r.data.result || {}),
+        reactionCount,
+        reactionId
+      }
+    }
+  }));
```

Comment:
actually the API was not returning any data after `/delete` call, and we needed to update the state. So I hacked its way into response (not ideal).

Another approach is to pass it directly to the [main mutation](https://github.com/hicommonwealth/commonwealth/blob/33b9792cba813e21041903d5b22ec543d491e11f/packages/commonwealth/client/scripts/state/api/comments/deleteReaction.ts#L52), but I worry it will cause a lot of bloatware (variables everywhere) leading in the future. 

I think for `/delete` API calls, there should be a minimum response, maybe the safe-to-return data like deleted resource ids would be nice (but I also think /delete calls should not return anything). I am a bit divided here, curious as to what you think.

Reply:
what you are saying is another thing I was not mentioning here. Not sure what is our pattern for returning data from backend but either we stick to the rule that DELETE does not returned anything but 'success' or we can adjust backend to return what we need on the front end. It's up to you how you handle it.

What I meant is that if we use `await`, let's don't use `then` so the code would be like:

```
  const response = await axios.delete(
    `${app.serverUrl()}/reactions/${reactionId}`,
    {
      data: {
        jwt: app.user.jwt,
        canvas_action: action,
        canvas_session: session,
        canvas_hash: hash,
      },
    }
  );

  return {
    ...response,
    data: {
      ...response.data,
      result: {
        ...(response.data.result || {}),
        reactionCount,
        reactionId,
      },
    },
  };
```

Reply:
Done